### PR TITLE
docs: Added written proposal process.

### PR DESCRIPTION
**Pull request by bwplotka**
Signed-off-by: Bartlomiej Plotka <bwplotka@gmail.com>

### CODE REVIEW

File: docs/contributing/proposal-process.md [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+If there is no problem, there is no need for changing anything, no need for a proposal. This might feel trivial, but we should first ask ourselves this question before even thinking about writing a proposal.
+
+It takes time to propose an idea, find consensus and implement more significant concepts, so let's not use the time before we feel it's worth it. Even good ideas sometimes have to wait for a good moment to discuss them.
+
+Let's assume the idea sounds interesting to you; what to do next, where to propose it? How to review it? Follow the algorithm below:
+
+![where](../img/thanos_proposal_flow.png)
+
+> Note: It's totally ok for your proposal to be rejected if the community feels the idea is not needed now or having gaps. Not all ideas are as good as it looks at the first glance, with less context. Sometimes it's not good time for it either.It's better to explicitly oppose it than to ignore it and leave it in limbo.
```
Comment (wiardvanrij) [inline docs/contributing/proposal-process.md:17]:
```suggestion
> Note: It's totally ok for your proposal to be rejected if the community feels the idea is not needed now or having gaps. Not all ideas are as good as it looks at the first glance, with less context. Sometimes it's not a good time for it either.It's better to explicitly oppose it than to ignore it and leave it in limbo.
```

File: docs/contributing/proposal-process.md [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+## Why we have Proposal Process?
```
Comment (wiardvanrij) [inline docs/contributing/proposal-process.md:19]:
```suggestion
## Why do we have a Proposal Process?
```

File: docs/contributing/proposal-process.md [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+* **Owners:**
+  * `<@author: single champion for the moment of writing>`
+
+* **Related Tickets:**
+  * `<JIRA, GH Issues>`
+
+* **Other docs:**
+  * `<Links…>`
+
+> TL;DR: Give here a short summary of what this document is proposing and what components it is touching.
```
Comment (wiardvanrij) [inline docs/contributing/proposal-process.md:36]:
```suggestion
> TL;DR: Give a summary of what this document is proposing and what components it is touching.
```

File: docs/contributing/proposal-process.md [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+>
+> *For example: This design doc is proposing a consistent design template for “example.com” organization.*
+
+## Why
+
+Put here a motivation behind the change proposed by this design document, give context.
```
Comment (wiardvanrij) [inline docs/contributing/proposal-process.md:42]:
```suggestion
Give a motivation behind the change proposed by this design document, provide context.
```

File: docs/contributing/proposal-process.md [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+
+Goals and use cases for the solution as proposed in [How](#how):
+
+* Allow easy collaboration and decision making on design ideas.
+* Have a consistent design style that is readable and understandable.
+* Have a design style that is concise and covers all the essential information.
+
+### Audience
+
+If not clear, the target audience that this change relates to.
```
Comment (wiardvanrij) [inline docs/contributing/proposal-process.md:62]:
```suggestion
If this is not clear already, provide the target audience for this change.
```

**Review by wiardvanrij — DISMISSED** [review]

File: docs/proposals-done/202106-proposals-process.md [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+  * [KEP Process](https://github.com/kubernetes/enhancements/blob/master/keps/README.md)
+  * [RHOBS Handbook Proposal Process](https://github.com/rhobs/handbook/pull/15)
+
+> TL;DR: We would like to propose an improved, official proposal process for Monitoring Group that clearly states when, where and how to create proposal/enhancement/design documents.
+
+## Why
+
+More extensive architectural, process, or feature decisions are hard to explain, understand and discuss. It takes a lot of time to describe the idea, to motivate interested parties to review it, give feedback and approve. That's why it is essential to streamline the proposal process.
+
+Given that as open source community we work in highly distributed teams and work with multiple communities, we need to allow asynchronous discussions. This means it's essential to structure the talks into shared documents. Persisting in those decisions, once approved or rejected, is equally important, allowing us to understand previous motivations.
```
Comment (GiedriusS) [inline docs/proposals-done/202106-proposals-process.md:24]:
```suggestion
Given that as open-source community we work in highly distributed teams and work with multiple communities, we need to allow asynchronous discussions. This means it's essential to structure the talks into shared documents. Persisting those decisions, once approved or rejected, is equally important, allowing us to understand previous motivations.
```

File: docs/proposals-done/202106-proposals-process.md [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+
+* **Other docs:**
+  * [KEP Process](https://github.com/kubernetes/enhancements/blob/master/keps/README.md)
+  * [RHOBS Handbook Proposal Process](https://github.com/rhobs/handbook/pull/15)
+
+> TL;DR: We would like to propose an improved, official proposal process for Monitoring Group that clearly states when, where and how to create proposal/enhancement/design documents.
+
+## Why
+
+More extensive architectural, process, or feature decisions are hard to explain, understand and discuss. It takes a lot of time to describe the idea, to motivate interested parties to review it, give feedback and approve. That's why it is essential to streamline the proposal process.
```
Comment (GiedriusS) [inline docs/proposals-done/202106-proposals-process.md:22]: Maybe we can vaguely describe _when_ such a design doc is needed? I think describing this should be pretty much necessary given that we want to avoid bureaucracy as much as possible, right?

Comment (bwplotka) [reply docs/proposals-done/202106-proposals-process.md:22]: I presented that in the flow, but I can repeat here too 🤗

Comment (bwplotka) [reply docs/proposals-done/202106-proposals-process.md:22]: Added section to proposal process

**Comment by bwplotka** [comment]
Lol, no. That's why I think we need to make sure the process is defined better ;p

**Comment by bwplotka** [comment]
I will adjust it with new info

**Comment by bwplotka** [comment]
ignore previous comments (was for different PR) (:

**Comment by bwplotka** [comment]
PTAL @thanos-io/thanos-maintainers

**Review by Namanl2001 — APPROVED** [review]
LGTM 👍

**Comment by bwplotka** [comment]
What is a TLDR (: 

PTAL @GiedriusS @squat @onprem does it make sense?

**Review by GiedriusS — APPROVED** [review]
:+1: :muscle: awesome!
